### ISSUE TITLE
using catkin on plain cmake packages

### ISSUE THREAD

**Issue by smits** [ros/catkin#362]
Hi I'm using catkin to build my plain cmake packages. I added a package.xml file in these packages and instrumented it with the highly undocumented build_type export tag.

``` xml
 <export>
      <build_type>                                                                                                                                                                                                  
        cmake                                                                                                                                                                                                       
      </build_type>
 </export>
```

My workspace looks as follows:

```
~~  traversing packages in topological order:
~~  - catkin
~~  - log4cpp (plain cmake)
~~  - rtt (plain cmake)
~~  - ocl (plain cmake)
~~  - utilmm (plain cmake)
~~  - typelib (plain cmake)
~~  - rtt_typelib (plain cmake)
```

catkin is there because I need at least commit b17de46d, I have catkin at fb7b1c6f55a, because 6a3a9f22f46 is breaking stuff for me.

I use catkin_make_isolated and all goes well until I get to the ocl package:

==> Processing plain cmake package: 'ocl'
==> Creating build directory: 'build_isolated/ocl'
==> Building with env: '/home/rsmits/ros_orocos_catkin_ws/devel_isolated/rtt/env.sh'
==> cmake /home/rsmits/ros_orocos_catkin_ws/src/orocos_toolchain/ocl -DCMAKE_INSTALL_PREFIX=/home/rsmits/ros_orocos_catkin_ws/devel_isolated/ocl
/home/rsmits/ros_orocos_catkin_ws/devel_isolated/rtt/env.sh: 5: .: Can't open /home/rsmits/ros_orocos_catkin_ws/devel_isolated/log4cpp/setup.sh

As expected there is indeed no /home/rsmits/ros_orocos_catkin_ws/devel_isolated/log4cpp/setup.sh but I do have a /home/rsmits/ros_orocos_catkin_ws/devel_isolated/log4cpp/env.sh

Is this just a naming mismatch or is there more going on?

**Comment by smits** [comment]
If I update the builder.py script as follows:

```
diff --git a/python/catkin/builder.py b/python/catkin/builder.py
index 26c66fc..6a9f96e 100644
--- a/python/catkin/builder.py
+++ b/python/catkin/builder.py
@@ -319,7 +319,7 @@ def build_cmake_package(
     # Generate basic env.sh for chaining to catkin packages
     new_env_path = os.path.join(install_target, 'env.sh')
     subs = {}
-    subs['last_env'] = os.path.join(os.path.dirname(last_env), 'setup.sh') if last_env is not None else None
+    subs['last_env'] = os.path.join(os.path.dirname(last_env), 'env.sh') if last_env is not None else None
     subs['cmake_prefix_path'] = install_target + ":"
     subs['ld_path'] = os.path.join(install_target, 'lib') + ":"
     pythonpath = ":".join(get_python_path(install_target))
```

It is still not working, apparently the devel install spaces of the plain cmake packages are not added correctly to the cmake_prefix_path of the depending packages.

**Comment by dirk-thomas** [comment]
Related to #340. Was fixed in b17de46d990209b2360eff02db2d4416ff6fdd69.
